spring-integration-file/src/main/java/org/springframework/integration/file/remote/synchronizer/AbstractInboundFileSynchronizingMessageSource.java

```diff
@@ -144,6 +144,10 @@ public void afterPropertiesSet() throws Exception {
 					throw new FileNotFoundException(this.localDirectory.getName());
 				}
 			}
+			this.fileSource.setUseWatchService(true);
```

Comment:
Should this be an option?

Watch events are slow on OS X - this means using the `delete()` to reset the filter might not be fast enough.

We would also need to keep the current documentation for people who can't rely on the watch service (manual removal from filter on rollback).